tests/h/services/group_test.py

```diff
@@ -433,32 +483,50 @@ def test_provides_realtime_publisher_as_publish(self, patch, pyramid_request):
         })
 
 
-class GroupScopeWithOrigin(Matcher):
-    """Matches any GroupScope with the given origin."""
+@pytest.fixture
+def usr_svc(pyramid_request, db_session):
+    def fetch(userid):
+        # One doesn't want to couple to the user fetching service but
+        # we do want to be able to fetch user models for internal
+        # module behavior tests
+        return db_session.query(User).filter_by(userid=userid).one_or_none()
+    return fetch
 
-    def __init__(self, origin):
-        self.origin = origin
 
-    def __eq__(self, group_scope):
-        if not isinstance(group_scope, GroupScope):
-            return False
-        return group_scope.origin == self.origin
+@pytest.fixture
+def origins():
+    return ['http://example.com']
+
+
+@pytest.fixture
+def publish():
+    return mock.Mock()
+
+
+@pytest.fixture
+def svc(db_session, usr_svc, publish):
+    return GroupService(db_session, usr_svc, publish=publish)
+
+
+@pytest.fixture
+def creator(factories):
+    return factories.User(username='group_creator')
```

Comment:
Is `creator` the right name for this fixture? At the point where this fixture has run and the `creator` argument has been passed into the test method it's just a user and not actually the creator of anything. It only becomes the creator when the test itself passes it as the creator to a create group method.

Many of the methods that use the `creator` fixture just pass it to a create group method and then the test doesn't use `creator` again after that. So the test could just pass `creator=factories.User().userid` to the method instead.